kube/utils.go

```diff
@@ -75,6 +75,23 @@ func (b *ConfigBuilder) SetAPIVersion(apiVersion string) *ConfigBuilder {
 	return b
 }
 
+// SetConditionalAPIVersion sets the kube API version of the resource to build;
+// if that API version is not available, use a fallback instead (to be
+// compatible with older releases of kube).  If we are not building a helm
+// chart, the desired API version is always used.
+func (b *ConfigBuilder) SetConditionalAPIVersion(apiVersion, fallbackAPIVersion string) *ConfigBuilder {
+	if b.settings == nil || !b.settings.CreateHelmChart {
+		b.apiVersion = apiVersion
+	} else {
+		b.apiVersion = fmt.Sprintf(
+			`{{ ternary "%s" "%s" (.Capabilities.APIVersions.Has "%s") }}`,
```

Comment:
Do we document anywhere the minimum helm version required? I think `ternary` requires helm 2.10 (which seems sensible, even though we sometimes use 2.8.x).

Should we have a test in the generated charts to make sure the helm version is compatible?

Reply:
Looks like we're still testing against [2.8.2](https://github.com/SUSE/cf-ci/blob/develop/cf-ci-orchestration/Dockerfile#L41), hmm.

Reply:
Okay, so it turns out `helm template` does _not_ support `.Capabilities.APIVersions.Has` style checks, so we can't just run that to see if we have support for older clients.

Will need to get rid of `ternary`, I guess. (But yes, that doesn't work, though a good document on what versions we support, and figuring out how to test them, would be nice)

Reply:
Removed `ternary` in c642ece2c6e51fbd825c2e371a924a09ecdefcd4. Kind of annoyed that I can't actually test with `helm` in a test, though.